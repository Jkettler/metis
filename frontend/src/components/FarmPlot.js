import React, { Component } from 'react';
import Plot from './Plot';
import ProfitCalculator from './ProfitCalculator';

class FarmPlot extends Component {
  constructor(props){
    super(props);
    this.state = {
      crop: {},
      planted: {},
      allCrops: {}
    };
  }

  componentDidMount(){
    this.setState({
      crop: this.props.crop,
      allCrops: this.props.allCrops
    });
  }

  onChange(child_props){

    let { planted } = this.state;
    let { x, y, id } = child_props;

    let key = `${x}${y}`;

    if(planted[key] === undefined){
      planted[key] = id;
    } else {
      planted[key] = undefined;
    }

    this.setState({
      planted: planted
    });
  }

  render() {
    let dimension_x = 5;
    let dimension_y = 5;
    let { crop, allCrops } = this.props;
    let { planted } = this.state;

    return (
      <div>
        {new Array(dimension_y).fill(-1).map((el, i) =>
          <div className="Metis-game-row" key={i}>
            {new Array(dimension_x).fill(-1).map((el, j) =>
              <div className="Metis-game-column" key={j}>
                <Plot onChange={this.onChange.bind(this)} y={i} x={j} color={crop.ui_color} id={crop.id}/>
              </div>
            )}
          </div>
        )}
        <ProfitCalculator cropInfo={allCrops} total={planted}/>
      </div>
    )
  }
}export default FarmPlot;