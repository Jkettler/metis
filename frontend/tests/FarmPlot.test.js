import React from 'react';
import FarmPlot from '../src/components/FarmPlot';
import renderer from 'react-test-renderer';

test('does a render', () => {
  const allCrops = [{}];
  const crop = {};

  const component = renderer.create(
    <FarmPlot allCrops={allCrops} crop={crop}/>
  );
  let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});