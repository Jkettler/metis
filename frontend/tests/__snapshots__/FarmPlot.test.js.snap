// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does a render 1`] = `
<div>
  <div
    className="Metis-game-row"
  >
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
  </div>
  <div
    className="Metis-game-row"
  >
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
  </div>
  <div
    className="Metis-game-row"
  >
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
  </div>
  <div
    className="Metis-game-row"
  >
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
  </div>
  <div
    className="Metis-game-row"
  >
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <div
      className="Metis-game-column"
    >
      <span
        className="Metis-plot-square"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
  </div>
  <div
    className="Metis-profit-info"
  >
    Expected profit: $0
  </div>
</div>
`;
